lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -1,13 +1,26 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/buy_sell_par_shares'
+require_relative '../../../action/convert'
 
 module Engine
   module Game
     module G18OE
       module Step
         class BuySellParShares < Engine::Step::BuySellParShares
+          def setup
+            super
+
+            @can_convert = []
+            @game.corporations.each do |corp|
+              if corp.share_holders[current_entity] >= 50 && corp.type == :regional && @game.corporations.all?(&:ipoed)
+                @can_convert << corp
+              end
+            end
```

Comment:
As you're returning the items from the list you're iterating over, you could use `Array.select` or `Array.reject`.
```
@can_convert = @game.corporations.select do |corp|
  corp.share_holders[current_entity] >= 50 && corp.type == :regional && @game.corporations.all?(&:ipoed)
end
```